src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/AsyncCollectorBindingProvider.cs

```diff
@@ -0,0 +1,303 @@
+            {
+                var cm = parent._converterManager;
+                var patternMatcher = parent._patternMatcher;
+
+                var parameter = context.Parameter;
+                TAttribute attributeSource = parameter.GetCustomAttribute<TAttribute>(inherit: false);
+
+                Func<TAttribute, Task<TAttribute>> hookWrapper = null;
+                if (parent.PostResolveHook != null)
+                {
+                    hookWrapper = (attrResolved) => parent.PostResolveHook(attrResolved, parameter, parent._nameResolver);
+                }
+
+                var cloner = new AttributeCloner<TAttribute>(attributeSource, context.BindingDataContract, parent._nameResolver, hookWrapper);
+
+                Func<object, object> buildFromAttribute;
+                FuncConverter<TMessage, TAttribute, TType> converter = null;
+
+                // Prefer the shortest route to creating the user type.
+                // If TType matches the user type directly, then we should be able to directly invoke the builder in a single step. 
+                //   TAttribute --> TUserType
+                var checker = ConverterManager.GetTypeValidator<TType>();
+                if (checker.IsMatch(typeof(TMessage)))
+                {
+                    buildFromAttribute = patternMatcher.TryGetConverterFunc(
+                        typeof(TAttribute), typeof(IAsyncCollector<TMessage>));
+                }
+                else
+                {
+                    // Try with a converter
+                    // Find a builder for :   TAttribute --> TType
+                    // and then couple with a converter:  TType --> TParameterType
+                    converter = cm.GetConverter<TMessage, TType, TAttribute>();
+                    if (converter == null)
+                    {
+                        // Preserves legacy behavior. This means we can only have 1 async collector.
+                        // However, the collector's builder object can switch. 
+                        throw NewMissingConversionError(typeof(TMessage));
+                    }
+
+                    buildFromAttribute = patternMatcher.TryGetConverterFunc(
+                        typeof(TAttribute), typeof(IAsyncCollector<TType>));
+                }
+
+                if (buildFromAttribute == null)
+                {
+                    return null;
+                }
+
+                ParameterDescriptor param;
+                if (parent.BuildParameterDescriptor != null)
+                {
+                    param = parent.BuildParameterDescriptor(attributeSource, parameter, parent._nameResolver);
+                }
+                else
+                {
+                    param = new ParameterDescriptor
+                    {
+                        Name = parameter.Name,
+                        DisplayHints = new ParameterDisplayHints
+                        {
+                            Description = "input"
```

Comment:
Where does this "input" show up in the dashboard? Other places (e.g. BindCollector) we're using "output". What should it be?